### PR TITLE
[Docs]: fix error in documentation for r/aws_lambda_function

### LINKED ISSUE #29109: [Docs]: lambda_function resource documentation doesn't mention exactlyOneOf relation between filename, s3_bucket and image_uri
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#filename

### Description

The documentation states that s3_object, filename and image_uri conflicts with each other. But this behavior was changed on https://github.com/hashicorp/terraform-provider-aws/commit/bd986c7ab162bd319e4c1b508a2814a0d76d892c to make it compliant with aws documentation. However the documentation here has not been updated. The documentation should specify the exactlyOneOf nature of the relation instead of the conflictsWith as it does at the moment.

### References

Link to the commit : https://github.com/hashicorp/terraform-provider-aws/commit/bd986c7ab162bd319e4c1b508a2814a0d76d892c
 

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by hsiam261**
### Description
This PR addresses the issue https://github.com/hashicorp/terraform-provider-aws/issues/29109.
https://github.com/hashicorp/terraform-provider-aws/commit/bd986c7ab162bd319e4c1b508a2814a0d76d892c changed behavior of the aws lambda function resource, but that change was not reflected in the provider documentation. Since, that commit actually adheres to the original aws documentation, we need to update the provider docs.

### Relations

Closes #29109 

### References

Related commit: https://github.com/hashicorp/terraform-provider-aws/commit/bd986c7ab162bd319e4c1b508a2814a0d76d892c
### Output from Acceptance Testing

Documentation changes, not necessary.

### CODE REVIEW

File: website/docs/r/lambda_function.html.markdown [hunk 1]
```diff
@@ -265,19 +265,19 @@ The following arguments are optional:
 * `environment` - (Optional) Configuration block. Detailed below.
 * `ephemeral_storage` - (Optional) The amount of Ephemeral storage(`/tmp`) to allocate for the Lambda Function in MB. This parameter is used to expand the total amount of Ephemeral storage available, beyond the default amount of `512`MB. Detailed below.
 * `file_system_config` - (Optional) Configuration block. Detailed below.
-* `filename` - (Optional) Path to the function's deployment package within the local filesystem. Conflicts with `image_uri`, `s3_bucket`, `s3_key`, and `s3_object_version`.
+* `filename` - (Optional) Path to the function's deployment package within the local filesystem. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified.
```
Comment (justinretzolk) [inline website/docs/r/lambda_function.html.markdown:268]: I think "or" feels better than "and" here. Oxford comma, love it or hate it 🙂.

```suggestion
* `filename` - (Optional) Path to the function's deployment package within the local filesystem. Exactly one of `filename`, `image_uri`, or `s3_bucket` must be specified.
```

File: website/docs/r/lambda_function.html.markdown [hunk 2]
```diff
@@ -265,19 +265,19 @@ The following arguments are optional:
...
 * `handler` - (Optional) Function [entrypoint][3] in your code.
 * `image_config` - (Optional) Configuration block. Detailed below.
-* `image_uri` - (Optional) ECR image URI containing the function's deployment package. Conflicts with `filename`, `s3_bucket`, `s3_key`, and `s3_object_version`.
+* `image_uri` - (Optional) ECR image URI containing the function's deployment package. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified.
```
Comment (justinretzolk) [inline website/docs/r/lambda_function.html.markdown:271]: See comment on line 268.

```suggestion
* `image_uri` - (Optional) ECR image URI containing the function's deployment package. Exactly one of `filename`, `image_uri`,  or `s3_bucket` must be specified.
```

File: website/docs/r/lambda_function.html.markdown [hunk 3]
```diff
@@ -265,19 +265,19 @@ The following arguments are optional:
...
 * `publish` - (Optional) Whether to publish creation/change as new Lambda Function Version. Defaults to `false`.
 * `reserved_concurrent_executions` - (Optional) Amount of reserved concurrent executions for this lambda function. A value of `0` disables lambda from being triggered and `-1` removes any concurrency limitations. Defaults to Unreserved Concurrency Limits `-1`. See [Managing Concurrency][9]
 * `runtime` - (Optional) Identifier of the function's runtime. See [Runtimes][6] for valid values.
-* `s3_bucket` - (Optional) S3 bucket location containing the function's deployment package. Conflicts with `filename` and `image_uri`. This bucket must reside in the same AWS region where you are creating the Lambda function.
-* `s3_key` - (Optional) S3 key of an object containing the function's deployment package. Conflicts with `filename` and `image_uri`.
+* `s3_bucket` - (Optional) S3 bucket location containing the function's deployment package. This bucket must reside in the same AWS region where you are creating the Lambda function. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified. Specifying the `s3_key` is manditory when `s3_bucket` is specified.
```
Comment (justinretzolk) [inline website/docs/r/lambda_function.html.markdown:279]: Again re:line 268. Shortened the last sentence a bit and made it more neutral so that the same wording can be used below.

```suggestion
* `s3_bucket` - (Optional) S3 bucket location containing the function's deployment package. This bucket must reside in the same AWS region where you are creating the Lambda function. Exactly one of `filename`, `image_uri`, or `s3_bucket` must be specified. When `s3_bucket` is set, `s3_key` is required.
```

File: website/docs/r/lambda_function.html.markdown [hunk 4]
```diff
@@ -265,19 +265,19 @@ The following arguments are optional:
...
+* `s3_key` - (Optional) S3 key of an object containing the function's deployment package. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified. Specifying the `s3_bucket` is manditory when `s3_key` is specified.
```
Comment (justinretzolk) [inline website/docs/r/lambda_function.html.markdown:280]: Made the last sentence match above, for consistency. I don't think we need to mention the `filename`, `image_uri`, and `s3_bucket` bit here if we're going on relation to the schema. It feels a bit irrelevant when I read it, personally.

Given that you raised this PR, I'm interested in your thoughts on whether we can omit that bit. 

```suggestion
* `s3_key` - (Optional) S3 key of an object containing the function's deployment package. When `s3_bucket` is set, `s3_key` is required.
```

**Review by justinretzolk — COMMENTED** [review]
Hey @hsiam261 👋 Thank you for your contribution! I left a few notes that I'd like to look over together before we get this merged in. They're relatively small notes, but I figure while we're here, it's worth getting right 🙂

**Comment by hsiam261** [comment]
Hi @justinretzolk
Thanks for your suggestions. They surely improve my changes. So I've added your changes to this PR.

**Comment by hsiam261** [comment]
Hi @justinretzolk. Is there anything that's preventing this PR from getting merged?

**Comment by justinretzolk** [comment]
Hey @hsiam261 👋 Thank you for following up on this! It's been a hectic week, and this fell off of my radar a bit, but I'll get it merged in now! Thank you again for your contribution 🚀
